bounded_counter/count_min_sketch.py

```diff
@@ -17,71 +17,148 @@ def hash_key(row, key):
 class CountMinSketch(object):
     """
     Data structure used to estimate frequencies of elements in massive data sets with fixed memory footprint.
+
+Example:
+        {
+            cms = CountMinSketch(512) # Use 512 MB
+            print(cms.width)  # 16 777 216
+            print(cms.depth)   # 8
+            print(cms.size)  # 536 870 912 (512 MB in bytes)
+
+            cms.increment("foo")
```

Comment:
We agreed on Counter interface + raising `NotImplementedError` for methods that cannot be done using a given structure (key iteration etc).

Or is the left for the final "wrapper", which will use this `CountMinSketch` class internally? (=this class not directly meant for end users?)

Reply:
For CountMin-Sketch, trying to mimic the Counter interface doesn't really make sense. Even supporting the most traditional `cms['foo']+=1` is not possible(1) due to how python interprets such call (always as a successive __getitem__ -> __setitem__).

(1)You *could* work around that by writing a wrapper which returns custom objects instead of integers that delegate their __iadd__ to the structure but that sounds like a major overkill.